### ISSUE TITLE
Event Sources?

### ISSUE THREAD

**Issue by veqryn** [apex/apex#437]
I am trying to create lambda functions that use an event source, specifically "CloudWatch Events - Schedule", with a cron/schedule of "rate(1 hour)".

Though this can be done with the aws cli, I can't see how to do this with APEX, in your documentation.  

Is it possible right now?  And if not, could this be prioritized?

**Comment by tj** [comment]
Currently I recommend checking out Terraform (https://www.terraform.io/) for managing this sort of thing, it lets you preview changes made to your infrastructure etc, lots of things we probably won't duplicate. 

That said I would like to potentially support at least some of the basics to get people going quicker in the future since sources are so integral. I have some ideas that would make it pretty cool I think, just need to decide how much of Terraform I have time to duplicate haha.

I'll add a FAQ item for now so it's not just left out entirely!

**Comment by veqryn** [comment]
Thanks for looking.

For me, terraform is more for setting up infrastructure, rather than deploying my code to said infrastructure.
Choosing a particular lambda function's source, has a lot more to do with the business logic of the lambda function and its deployment, than it does with infrastructure setup, which is why I'd prefer it to live within the apex json configs.

**Comment by tj** [comment]
Yeah for me ideally it all lives in Apex, but with limited time that is hard. The more we put in apex(1) the more awkward the relationship between Terraform and Apex gets so I want to be careful there. I really prefer TF to CloudFormation but it may be worth switching (or adding) CF just to move this sort of thing into Apex a bit easier.

**Comment by veqryn** [comment]
I too prefer terraform over cloudformation.

What if you just required terraform installed, but still managed stuff with apex config files?
As the end user, I don't really care what you do in between me creating the apex config files + running apex deploy, and having my lambda functions show up...

Or alternatively, provide a hook to run specific terraform scripts after deploying the lambda, and then give a couple full examples of using said hook to setup an event source for the deployed lambda function.

**Comment by bhurlow** [comment]
If full implementation of event sources is a stretch, it might still be nice allow the apex config to determine the frequency that the function is executed – i.e. an event source with no data that just triggers the function routinely. A bonus would be a sane time format like `1m` or `1.5h`

**Comment by tj** [comment]
@bhurlow scheduled events you mean? 

I think if we were to add one, it would be best to add them all. I just don't want to go down that road unless it's all-inclusive, effectively removing the need for Terraform. I'll have to play around with CloudFormation some more, maybe it has improved lately.

**Comment by mazzy89** [comment]
@tj which is currently the state of the art regard the schedule of functions in apex?

**Comment by tj** [comment]
@mazzy89 still via Terraform at the moment

**Comment by garethr** [comment]
@tj do you know if there is a Terraform based example of doing this with Apex?
